src/scrapers/mizrahi.ts

```diff
@@ -130,6 +162,70 @@ function convertTransactions(txns: ScrapedTransaction[]): Transaction[] {
   });
 }
 
+async function getTransactionExtraScrap(record: ScrapedTransaction, headers: Headers): Promise<ExtraTransactionResult | null> {
+  const formattedPeulaDate = moment(record.MC02PeulaTaaEZ).format(DATE_FORMAT);
+  const data = {
+    inKodGorem: record.MC02KodGoremEZ,
+    inAsmachta: record.MC02AsmahtaMekoritEZ,
+    inSchum: record.MC02SchumEZ,
+    inNakvanit: record.MC02KodGoremEZ,
+    inSugTnua: record.MC02SugTnuaKaspitEZ,
+    inAgid: record.MC02AgidEZ,
+    inTarPeulaFormatted: formattedPeulaDate,
+    inTarErechFormatted: formattedPeulaDate,
+    inKodNose: record.MC02SeifMaralEZ,
+    inKodTatNose: record.MC02NoseMaralEZ,
+    inTransactionNumber: record.TransactionNumber,
+  };
+  
+  try {
+    const res = await fetchPost(TRANSACTION_DETAILS_REQUEST_URL, data, headers);
+    return res;
+  } catch (e) {
+    console.error(`Error fetching extra transaction details for record ${JSON.stringify(record)}`, e);
+  }
+  return null;
+}
+
+function simplifyExtraTransactionResultsToMemo(extraResult: ExtraTransactionResult): string {
+  let memo = '';
+  extraResult.body.fields.forEach(field => 
+    field?.forEach(group => 
+      group?.Records.forEach(record => 
+        record?.Fields.forEach((fieldRecord) => {
+          memo += `${fieldRecord.Label} ${fieldRecord.Value}; `;
+        }),
+      ),
+    ),
+  );
+  return memo;
+}
+
+async function getExtraScrap(originalRecords: ScrapedTransaction[], currentTxns: Transaction[], headers: Headers): Promise<Transaction[]> {
+  const recordsWithDetails = originalRecords
+    .map((record, index) => ({ record, index }))
+    .filter(({ record }) => record.MC02ShowDetailsEZ === '1');
+
+  const promises = recordsWithDetails.map(({ record }) => getTransactionExtraScrap(record, headers));
+  let accounts: Array<ExtraTransactionResult | null> = [];
+  while (promises.length > 0) {
+    const currentPromises = promises.splice(0, TRANSACTION_DETAILS_REQUEST_CONCURRENCY);
+    accounts = accounts.concat(await Promise.all(currentPromises));
+    await sleep(TRANSACTION_DETAILS_REQUEST_WAIT_TIME);
+  }
```

Comment:
I think you're trying to do a batch proccesing here, but you missed something.

Once you run an `async` function, it runs! When you called `getTransactionExtraScrap(record, headers)`, the function executed. It happened on the function **call** and not on the `await` call.

In your code, the `promises` hold **all the executions**, and you just **waiting** them one-by-one, it is not logical.